linux-puppet-agent/azuredeploy.json

```diff
@@ -0,0 +1,246 @@
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "allowedValues": [
+        "Standard_A1",
+        "Standard_A2",
+        "Standard_A3",
+        "Standard_A4",
+        "Standard_D1",
+        "Standard_D2",
+        "Standard_D3",
+        "Standard_D4"
+      ],
+      "metadata": {
+        "description": "Size of vm, e.g: Standard_D2"
+      },
+      "defaultValue": "Standard_D2"
+    },
+    "username": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the Virtual Machine, e.g: joetheadmin."
+      }
+    },
+    "password": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "newStorageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS prefix for the Storage Account where the Virtual Machine's disks will be placed."
+      }
+    },
+    "dnsNameForPublicIP": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "pupmstrFQDN": {
+      "type":"string",
+      "metadata": {
+        "description": "FQDN of your PE box, e.g: pupmstr.cloudapp.net"
+      },
+      "defaultValue": "pupmstr.cloudapp.net"
+    },
+    "pupmstrIP": {
+      "type":"string",
+      "metadata": {
+        "description": "IP of your PE box, e.g: 192.168.1.1"
+      },
+      "defaultValue": "192.168.1.1"
+    },
+    "pupmstrInternalFQDN": {
+      "type":"string",
+      "metadata": {
+        "description": "Internal FQDN of your PE box, e.g: pupmstr.pupmstr.d6.internal.cloudapp.net.  Get it from node requests page in PE Console."
+      },
+      "defaultValue": "pupmstr.pupmstr.d6.internal.cloudapp.net"
+    }
+  },
+  "variables": {
+    "extensionName": "CustomScript",
+    "scenarioPrefix": "puppetAgentLinux",
+    "apiVersion": "2015-06-15",
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "ubuntuOSVersion": "14.04.2-LTS",
+    "OSDiskName": "[concat(variables('scenarioPrefix'),'OSDisk')]",
+    "nicName": "[concat(variables('scenarioPrefix'),'Nic')]",
+    "vnetAddressPrefix": "10.0.0.0/16",
+    "subnetName": "[concat(variables('scenarioPrefix'),'Subnet')]",
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "[concat(variables('scenarioPrefix'),'PublicIp')]",
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "[concat(variables('scenarioPrefix'),'VM')]",
+    "virtualNetworkName": "[concat(variables('scenarioPrefix'),'Vnet')]",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "fileUris":"https://raw.githubusercontent.com/Azure/master/linux-puppet-agent/install_puppet_agent.sh",
```

Comment:
Should be: `https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/linux-puppet-agent/install_puppet_agent.sh`